event/event.go

```diff
@@ -0,0 +1,214 @@
+	// DeltaHorizontal is the change in Horizontal since the last MouseWheel event.
+	DeltaHorizontal float32
+}
+
+// MouseButtonDown is a mouse button press event.
+type MouseButtonDown struct {
+	// X position of the mouse pointer.
+	X float32
+	// Y position of the mouse pointer.
+	Y float32
+	// DeltaX is the change in X since the last mouse event.
+	DeltaX float32
+	// DeltaY is the change in Y since the last mouse event.
+	DeltaY float32
+	// Button that was pressed.
+	// TODO: this should change later from an int to an enumeration type
+	Button int
+	// Pressure applied on the mouse click.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// MouseButtonDown is a mouse button Release event.
+// The data is identical to a MouseButtonDown event.
+type MouseButtonUp MouseButtonDown
+
+// MouseEnter occurs when the mouse enters the view window.
+type MouseEnter struct {
+	// X position of the mouse pointer.
+	X float32
+	// Y position of the mouse pointer.
+	Y float32
+	// DeltaX is the change in X since last mouse event.
+	DeltaX float32
+	// DeltaY is the change in Y since last mouse event.
+	DeltaY float32
+}
+
+// MouseLeave occurs when the mouse leaves the view window.
+// The data is identical to MouseEnter
+type MouseLeave MouseEnter
+
+// ViewUpdate occurs when the application is ready to update
+// the next frame on the view port.
+type ViewUpdate struct {
+	// No data neccesary, for now.
+}
+
+// ViewSize occurs when the size of the application's view port changes.
+type ViewSize struct {
+	// Width is the actual, real width of the view.
+	Width float32
+	// Height is the actual, real height of the view.
+	Height float32
+	// X position of the view on the physical screen.
+	X float32
+	// Y position of the view on the physical screen.
+	Y float32
+}
+
+// TouchBegin occurs when a touch begins.
+type TouchBegin struct {
+	// ID of the touch that caused the touch event.
+	ID int
+	// X position of the event.
+	X float32
+	// Y position of the event.
+	Y float32
+	// Pressure of applied touch.
+	Pressure float32
+	// Primary represents whether the touch event is the primary touch or not.
+	Primary bool
+}
+
+// TouchMoved occurs when a touch moved, or in other words, is dragged.
+type TouchMoved struct {
+	// ID of the touch that caused the touch event.
+	ID int
+	// X position of the event.
+	X float32
+	// Y position of the event.
+	Y float32
+	// DeltaX is the change in X since last touch event.
+	DeltaX float32
+	// Deltay is the change in Y since last touch event.
+	DeltaY float32
+	// Pressure of applied touch.
+	Pressure float32
+	// Primary represents whether the touch event is the primary touch or not.
+	Primary bool
+}
+
+// TouchEnd occurs when a touch ends.
+// The data is the same as for a TouchMoved event.
+type TouchEnd TouchMoved
+
+// TouchCancel occurs when a touch is canceled.
+type TouchCancel struct {
+	// ID of the touch that caused the touch event.
+	ID int
```

Comment:
What's the difference between TouchEnd and TouchCancel BTW?

Reply:
On Android a touch can be cancelled. See http://stackoverflow.com/questions/11960861/what-causes-a-motionevent-action-cancel-in-android for some more details.  Similarly for iOS: https://developer.apple.com/documentation/uikit/uiresponder/1621116-touchescancelled . So let's keep this event.

Reply:
Ack.